src/backtrace.c

```diff
@@ -108,10 +108,16 @@ int backtrace_snapshot(int pid, int *tids, int *index, int nr_tids)
         return -1;
 
     for (i = 0; i < snap->num_threads; ++i) {
+        int x;
         char comm[16];
+        char end_pad[25] = "------------------------";
 
-        get_thread_comm(snap->tids[i], comm, sizeof(comm));
-        printf("--------------------  thread %d (%d) (%s)  --------------------\n", (index != NULL ? index[i] : i+1), snap->tids[i], comm);
+        x = get_thread_comm(snap->tids[i], comm, sizeof(comm));
+        if (x > 0 && x <= sizeof(end_pad))
+        {
+            end_pad[sizeof(end_pad) - x] = '\0';
+            printf("-------------- thread %d (%d) (%s) %s\n", (index != NULL ? index[i] : i+1), snap->tids[i], comm, end_pad);
```

Comment:
If get_thread_num() fails for some reason, or the name is too long, the header is not printed at all. This is quite wrong. In case of error it should be indicated in some simple way (like we do when a file is deleted). If the name is too long, we can truncate it.